### PR TITLE
[PARTIALLY LEAK INFORMED] Miscellaneous cleanup and documentation

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: include/quest_log.h [hunk 1]
```diff
@@ -8,10 +8,11 @@
...
 {
-    u8 localId;
-    u8 mapNum;
-    u8 mapGroup;
-    u8 animId;
+    // When command == 2, these fields have different meanings
+    u8 localId;  // cmd == 2: Pressed A/B, checked wild, held direction, took step
+    u8 mapNum;   // cmd == 2: Always set to 0
+    u8 mapGroup; // cmd == 2: Dpad direction
+    u8 animId;   // cmd == 2: Always set to 0
```
Comment (GriffinRichards) [inline include/quest_log.h:15]: Union?

File: include/global.fieldmap.h [hunk 2]
```diff
@@ -293,6 +293,15 @@ enum {
...
 #define PLAYER_AVATAR_FLAG_DASH       (1 << PLAYER_AVATAR_STATE_WATERING)
 
+enum {
+    PLAYER_AVATAR_GFX_NORMAL,
+    PLAYER_AVATAR_GFX_BIKE,
+    PLAYER_AVATAR_GFX_SURF,
+    PLAYER_AVATAR_GFX_FIELD_MOVE,
+    PLAYER_AVATAR_GFX_FISH,
+    PLAYER_AVATAR_GFX_VSSEEKER,
+};
```
Comment (GriffinRichards) [inline include/global.fieldmap.h:303]: These are the FRLG equivalent constants to RSE's `PLAYER_AVATAR_STATE`. The latter should be removed here altogether (their uses in the `PLAYER_AVATAR_FLAG_` defines are also incorrect, they should just be the numbers 0-7 there)

Comment (PikalaxALT) [reply include/global.fieldmap.h:303]: Right, I was going to get around to fixing that.

File: include/quest_log.h (hunk 1 above)
Comment (PikalaxALT) [reply include/quest_log.h:15]: Would be idiotic if it was.

File: include/global.fieldmap.h (hunk 2 above)
Comment (PikalaxALT) [reply include/global.fieldmap.h]: These are indeed two completely separate enums. Names of both have been fixed.

File: include/constants/flags.h [hunk 3]
```diff
@@ -1324,7 +1324,7 @@
 #define FLAG_SYS_FLASH_ACTIVE                                       (SYS_FLAGS + 0x6)
 #define FLAG_SYS_SPECIAL_WILD_BATTLE                                (SYS_FLAGS + 0x7)
 
-#define FLAG_0x808                                                  (SYS_FLAGS + 0x8)
+#define FLAG_QL_DEPARTED                                                  (SYS_FLAGS + 0x8)
```
Comment (GriffinRichards) [inline include/constants/flags.h:1327]: Alignment

File: berry_fix/payload/include/global.fieldmap.h [hunk 4]
```diff
@@ -224,8 +224,8 @@ struct ObjectEventGraphicsInfo
 #define PLAYER_AVATAR_FLAG_ACRO_BIKE  (1 << 2)
 #define PLAYER_AVATAR_FLAG_SURFING    (1 << 3)
 #define PLAYER_AVATAR_FLAG_UNDERWATER (1 << 4)
-#define PLAYER_AVATAR_FLAG_FIELD_MOVE (1 << 5)
-#define PLAYER_AVATAR_FLAG_FISHING    (1 << 6)
+#define PLAYER_AVATAR_FLAG_CONTROLLABLE (1 << 5)
+#define PLAYER_AVATAR_FLAG_FORCED    (1 << 6)
```
Comment (GriffinRichards) [inline berry_fix/payload/include/global.fieldmap.h:228]: Alignment

File: include/constants/vars.h [hunk 5]
```diff
@@ -122,7 +122,7 @@
 #define VAR_EGG_BRAG_STATE                 0x404A
 #define VAR_LINK_WIN_BRAG_STATE            0x404B
 #define VAR_POKELOT_RND2                   0x404C
-#define VAR_0x404D                         0x404D
+#define VAR_QL_ENTRANCE                         0x404D
```
Comment (GriffinRichards) [inline include/constants/vars.h:125]: Alignment

File: include/field_weather.h [hunk 6]
```diff
@@ -35,8 +35,8 @@ struct Weather
     u8 fadeScreenCounter;
     bool8 readyForInit;
     u8 taskId;
-    u8 unknown_6CA;
-    u8 unknown_6CB;
+    u8 fade_in_active;
+    u8 fade_in_counter;
```
Comment (GriffinRichards) [inline include/field_weather.h:39]: `fadeInActive`, `fadeInCounter`

File: include/quest_log.h [hunk 7]
```diff
@@ -54,15 +55,18 @@ void *QuestLogGetFlagOrVarPtr(bool8 isFlag, u16 idx);
...
 void SetQuestLogRecordAndPlaybackPointers(void *oldSave);
 void sub_811246C(struct Sprite *sprite);
-void sub_81124EC(u8 a0, u8 a1, u8 a2, u8 a3);
+void QuestLogRecordNPCStep(u8 a0, u8 a1, u8 a2, u8 a3);
 bool8 sub_8111C2C(void);
-void sub_81128BC(u8 a0);
-void sub_811278C(u8 movementActionId, u8 duration);
+
+#define QL_ESCALATOR_OUT 1
+#define QL_ESCALATOR_IN  2
```
Comment (GriffinRichards) [inline include/quest_log.h:62]: Move the defines out of the declarations

File: src/field_player_avatar.c [hunk 8]
```diff
@@ -78,17 +78,17 @@ static void PlayerAvatarTransition_ReturnToField(struct ObjectEvent * playerObje
 static bool8 PlayerIsAnimActive(void);
 static bool8 PlayerCheckIfAnimFinishedOrInactive(void);
 static bool8 player_is_anim_in_certain_ranges(void);
-static bool8 sub_805BF58(void);
+static bool8 player_is_anim_in_certain_ranges__running_state_not_turn(void);
```
Comment (GriffinRichards) [inline src/field_player_avatar.c:81]: This is too long, vague, and not well formatted

File: src/field_player_avatar.c [hunk 9]
```diff
@@ -78,17 +78,17 @@ static void PlayerAvatarTransition_ReturnToField(struct ObjectEvent * playerObje
...
 static bool8 MetatileAtCoordsIsWaterTile(s16 x, s16 y);
-static void sub_805CC40(struct ObjectEvent * playerObjEvent);
+static void HandleWarpArrowSpriteHideShow(struct ObjectEvent * playerObjEvent);
 static void StartStrengthAnim(u8 objectEventId, u8 direction);
 static void Task_BumpBoulder(u8 taskId);
-static bool8 sub_805CD64(struct Task * task, struct ObjectEvent * playerObj, struct ObjectEvent * boulderObj);
+static bool8 do_boulder_init(struct Task * task, struct ObjectEvent * playerObj, struct ObjectEvent * boulderObj);
 static bool8 do_boulder_dust(struct Task * task, struct ObjectEvent * playerObj, struct ObjectEvent * boulderObj);
-static bool8 sub_805CE20(struct Task * task, struct ObjectEvent * playerObj, struct ObjectEvent * boulderObj);
+static bool8 do_boulder_finish(struct Task * task, struct ObjectEvent * playerObj, struct ObjectEvent * boulderObj);
```
Comment (GriffinRichards) [inline src/field_player_avatar.c:91]: `DoBoulderInit`, `DoBoulderFinish`, `DoBoulderDust`

File: src/field_player_avatar.c [hunk 10]
```diff
@@ -677,15 +678,15 @@ void SetPlayerAvatarTransitionFlags(u16 flags)
...
     [PLAYER_AVATAR_STATE_MACH_BIKE]  = PlayerAvatarTransition_Bike,
     [PLAYER_AVATAR_STATE_ACRO_BIKE]  = PlayerAvatarTransition_Bike,
     [PLAYER_AVATAR_STATE_SURFING]    = PlayerAvatarTransition_Surfing,
     [PLAYER_AVATAR_STATE_UNDERWATER] = PlayerAvatarTransition_Underwater,
-    [PLAYER_AVATAR_STATE_FIELD_MOVE] = PlayerAvatarTransition_ReturnToField,
-    [PLAYER_AVATAR_STATE_FISHING]    = PlayerAvatarTransition_Dummy,
-    [PLAYER_AVATAR_STATE_WATERING]   = PlayerAvatarTransition_Dummy
+    [PLAYER_AVATAR_STATE_CONTROLLABLE] = PlayerAvatarTransition_ReturnToField,
+    [PLAYER_AVATAR_STATE_FORCED]    = PlayerAvatarTransition_Dummy,
+    [PLAYER_AVATAR_STATE_DASH]   = PlayerAvatarTransition_Dummy
```
Comment (GriffinRichards) [inline src/field_player_avatar.c:689]: Alignment

File: src/field_weather.c [hunk 11]
```diff
@@ -127,7 +127,7 @@ static const u8 sBasePaletteGammaTypes[32] = {
     GAMMA_NORMAL,
 };
 
-const u16 gUnknown_83C2CE0[] = INCBIN_U16("graphics/field_effects/unk_83C2CE0.gbapal");
+const u16 gDefaultWeatherSpritePalette[] = INCBIN_U16("graphics/field_effects/unk_83C2CE0.gbapal");
```
Comment (GriffinRichards) [inline src/field_weather.c:130]: Name and move the file

File: src/metatile_behavior.c [hunk 12]
```diff
@@ -997,7 +997,7 @@ bool8 MetatileBehavior_IsBlinkingLights(u8 metatileBehavior)
         return FALSE;
 }
 
-bool8 MetatileBehavior_IsMB9F(u8 metatileBehavior)
+bool8 MetatileBehavior_IsIsNeatlyLinedUpTools(u8 metatileBehavior)
 {
     if(metatileBehavior == MB_9F)
```
Comment (GriffinRichards) [inline src/metatile_behavior.c:1002]: Could name this and the `MB_A3` behaviors too

File: include/constants/pokemon.h [hunk 13]
```diff
@@ -204,9 +204,10 @@
 #define FRIENDSHIP_EVENT_LEAGUE_BATTLE        0x3
 #define FRIENDSHIP_EVENT_LEARN_TMHM           0x4
 #define FRIENDSHIP_EVENT_WALKING              0x5
-#define FRIENDSHIP_EVENT_FAINT_SMALL          0x6
-#define FRIENDSHIP_EVENT_FAINT_OUTSIDE_BATTLE 0x7
-#define FRIENDSHIP_EVENT_FAINT_LARGE          0x8
+#define FRIENDSHIP_EVENT_MASSAGE              0x6
+#define FRIENDSHIP_EVENT_FAINT_SMALL          0x7
+#define FRIENDSHIP_EVENT_FAINT_OUTSIDE_BATTLE 0x8
+#define FRIENDSHIP_EVENT_FAINT_LARGE          0x9
```
Comment (GriffinRichards) [inline include/constants/pokemon.h:210]: These don't need to be hex

File: src/field_player_avatar.c (hunk 8 above)
Comment (PikalaxALT) [reply src/field_player_avatar.c:81]: Any suggestions on what to call it then?

Comment (GriffinRichards) [reply src/field_player_avatar.c:81]: I don't. If you don't have an actual name for it leave it as a `sub`

**Pull request merged**